lib/core/CDataFrame.cc

```diff
@@ -562,7 +597,11 @@ CDataFrame::sequentialApplyToAllRows(std::size_t beginRows,
         break;
     }
 
-    return {{std::move(func)}, true};
+    TRowFuncVec funcs;
+    funcs.reserve(1);
+    funcs.emplace_back(std::move(func));
+
+    return TRowFuncVecBoolPr{std::move(funcs), true};
```

Comment:
It would be worth adding a comment about why it's been done like this so that somebody doesn't simplify it in the future.

Reply:
I added an explanation in this [06f7e07](https://github.com/elastic/ml-cpp/pull/1142/commits/06f7e07f4b2cbcd6cc0553c381e4f91d116d12e2) commit.